### DISCUSSION TITLE
Catching error from $os.cmd().run()?

### DISCUSSION THREAD

**Discussion by AfdhalAtiffTan** [pocketbase/pocketbase#5364] [Q&A] [answered]
Hello,

Is it possible to catch an error returned by `$os.cmd('my-app').run()`?

I tried using `try...catch` block but it doesn't appear to catch it:
```js
try {
    $os.cmd('my-app').run();
} catch (error) {
    console.error(error);
}
```

Thank you.

**Comment by ganigeorgiev** [comment]
`$os.cmd` returns a pointer to the std Go [Cmd](https://pkg.go.dev/os/exec#Cmd) and every method attached to it is direct translations to its Go equivalent, aka. `run` will throw an error when [`Cmd.Run`](https://pkg.go.dev/os/exec#Cmd.Run) returns a non-nil error:

> The returned error is nil if the command runs, has no problems copying stdin, stdout, and stderr, and exits with a zero exit status.
>
> If the command starts but does not complete successfully, the error is of type [*ExitError](https://pkg.go.dev/os/exec#ExitError). Other error types may be returned for other situations.

**Reply by AfdhalAtiffTan** [reply]
Thank you @ganigeorgiev for your reply.

Forgive my ignorance, I am not familiar with Go yet.

Is it possible to catch the thrown error via a js hook?

At the moment I can only see the exit error in the Pocketbase's logs.

Thanks again.

**Reply by ganigeorgiev** [reply, marked as answer]
I'm confused and not sure that I understand what do you mean by _"I can only see the exit error in the Pocketbase's logs"_.

If you can see your `console.error` then this means that the thrown error is catched. If you want to propagate it you can try to rethrow it, aka. add `throw error` inside your `catch` block.

If your command doesn't trigger the `catch` block at all then make sure that it doesn't satisfy the above cited requirements in order the `run` method to throw (e.g. it needs to exit with a non-zero status).

**Reply by AfdhalAtiffTan** [reply]
Hello @ganigeorgiev,

Thank you for your speedy reply.

The problem was my `try...catch` in a js hook doesn't seem to catch the thrown error by the `$os.cmd().run()`.

After digging deeper, it turned out that I have another `$os.cmd().run()` in another file that was outside of the `try...catch` block. 

Silly me 😅

Thank you for your time.
